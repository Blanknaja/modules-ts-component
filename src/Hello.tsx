/**
 * @class ExampleComponent
 */

 import * as React from "react";
  import styled from "styled-components";



 export type Props = { text: string };
 
 export default class ExampleComponent extends React.Component<Props> {
   render() {
     const { text } = this.props;
 
     return <TestStyle style={{ color: "red" }}>Hello {text} naja</TestStyle>;
   }
 }


const TestStyle = styled.div`
  width: 250px;
  height: 200px;
  background: purple;
`;