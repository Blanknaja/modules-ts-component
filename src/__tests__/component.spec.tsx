import * as React from "react";
import Hello from "../Hello";
// import renderer from "react-test-renderer";
import * as renderer from "react-test-renderer";

test("TEST SUIT", () => {
  const component = renderer.create(<Hello text="World" />);
  const testInstance = component.root;

  expect(testInstance.findByType(Hello).props.text).toBe("World");

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
