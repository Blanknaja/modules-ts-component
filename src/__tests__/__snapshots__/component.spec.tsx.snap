// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST Project 1`] = `
<div>
  <h1>
    My Title :
    Card
  </h1>
</div>
`;

exports[`TEST SUIT 1`] = `
<div
  style={
    Object {
      "background": "green",
      "color": "red",
    }
  }
>
  Text : 
  World
   
</div>
`;
